packages/dd-trace/src/config.js

```diff
@@ -959,19 +955,38 @@ class Config {
   }
 
   _setArray (obj, name, value) {
-    if (value === null || value === undefined) {
+    if (value == null) {
       return this._setValue(obj, name, null)
     }
 
     if (typeof value === 'string') {
-      value = value && value.split(',')
+      value = value?.split(',')
     }
 
     if (Array.isArray(value)) {
       this._setValue(obj, name, value)
     }
   }
 
+  _setSamplingRule (obj, name, value) {
+    if (value == null) {
+      return this._setValue(obj, name, null)
+    }
+
+    if (typeof value === 'string') {
+      value = value?.split(',')
```

Comment:
also btw i just realized (should have said that in the previous comment), if `typeof value === 'string'`, then how could value.split be undefined ? we know it's a string sooo it should have split right ?

Reply:
for example, no check is made here: 
![image](https://github.com/DataDog/dd-trace-js/assets/70517128/e2b8dfad-0918-4dbf-9e2b-de40fc57f6dc)

Reply:
true that